src-built-in/components/apitest/test1.js

```diff
@@ -0,0 +1,1538 @@
+	windowClient.getComponentState({}, function (err, state) {
+		if (!err) {
+			alert('Sussceed. See console for detail.')
+			console.log(state)
+		}
+	})
+}
+
+function triggerGetBounds() {
+	windowClient.getBounds(function (err, bounds) {
+		if (!err) {
+			alert('Sussceed. See console for detail.')
+			console.log(bounds)
+		}
+	})
+}
+
+function triggerFitToDom() {
+	windowClient.fitToDOM({
+		maxHeight: 300,
+		maxWidth: 300,
+		padding: {
+			height: 100,
+			wdith: 100
+		}
+	}, function (err) {
+		if (!err) {}
+	})
+}
+
+function triggerCloseWindow() {
+	windowClient.close(false);
+}
+
+function triggerCancelTilingOrTabbing() {
+	windowClient.cancelTilingOrTabbing({
+		windowIdentifier: windowClient.getWindowIdentifier()
+	}, function (err) {
+		if (!err) {
+			alert('Sussceed.')
+		}
+	})
+}
+
+function triggerBringWindowToFront() {
+	windowClient.bringWindowToFront()
+}
+
+/* Storage client functions */
+function triggerGetStorageValue() {
+	storageClient.get({
+		topic: "finsemble",
+		key: "testKey"
+	}, function (err, val) {
+		if (!err) {
+			alert('Topic: finsemble, Key: testKey, value: ' + val)
+		}
+	})
+}
+
+function triggerGetStorageKeys() {
+	storageClient.keys({
+		topic: "finsemble"
+	}, function (err, keys) {
+		if (!err) {
+			alert('Successed. Check console for detail.')
+			console.log(keys)
+		}
+	})
+}
+
+function triggerRemoveStorageValue() {
+	storageClient.remove({
+		topic: "finsemble",
+		key: "testKey"
+	}, function (err) {
+		if (!err)
+			alert('Sussceed. Check console for detail.')
+	})
+}
+
+function triggerSaveStorageValue() {
+	if (document.getElementById('symbolInput').value != '')
+		storageClient.save({
+			topic: "finsemble",
+			key: "testKey",
+			value: document.getElementById('symbolInput').value
+		}, function (err) {
+			if (!err)
+				alert('Sussceed. Check console for detail.')
+		})
+	else
+		alert('Input a value.')
+}
+
+function triggerSetStorageStore() {
+	storageClient.setStore({
+		topic: "local",
+		dataStore: "LocalStorageAdapter"
+	}, function (err) {
+		if (!err)
+			alert('Sussceed. Check console for detail.')
+	})
+}
+
+function triggerSetStorageUser() {
+	storageClient.setUser({
+		user: 'testUser'
+	}, function (err) {
+		if (!err)
+			alert('Sussceed. Check console for detail.')
+	})
+}
+
+/* Distributed store client functions */
+function triggerSetStoreValue(field) {
+	distributedStoreClient.getStore({
+		store: 'testDs1',
+	}, function (err, store) {
+		if (!err) {
+			store.setValue({
+				field: field,
+				value: document.getElementById('symbolInput').value
+			}, function (err) {})
+		} else
+			alert('No such store.')
+	})
+}
+
+function triggerGetStoreValue() {
+	distributedStoreClient.getStore({
+		store: 'testDs1',
+	}, function (err, store) {
+		if (!err) {
+			store.getValue({
+				field: 'field1'
+			}, function (err, value) {
+				alert('Field1: ' + value)
+			})
+		} else
+			alert('No such store.')
+	})
+}
+
+function triggerRemoveStore() {
```

Comment:
The order of the test may need to be rearranged. If I start at the top and work my way down, the store is added and removed before I start adding/removing values from the store.